cache.go

```diff
@@ -51,25 +51,24 @@ var _ QuotaCache = (*RedisCache)(nil)
 var _ QuotaCache = (*LRU)(nil)
 var _ UsageCache = (*RedisCache)(nil)
 
-func NewLimitCounter(cfg RedisConfig, logger logger.Logger) httprate.LimitCounter {
+func NewLimitCounter(cfg RedisConfig, log logger.Logger) httprate.LimitCounter {
 	if !cfg.Enabled {
 		return nil
 	}
+	if log == nil {
+		log = logger.NewLogger(logger.LogLevel_INFO)
+	}
 	return httprateredis.NewCounter(&httprateredis.Config{
 		Host:      cfg.Host,
 		Port:      cfg.Port,
 		MaxIdle:   cfg.MaxIdle,
 		MaxActive: cfg.MaxActive,
 		DBIndex:   cfg.DBIndex,
 		OnError: func(err error) {
-			if logger != nil {
-				logger.Error("redis counter error", slog.Any("error", err))
-			}
+			log.With(slog.Any("error", err)).Error("redis counter error")
```

Comment:
`slog.With()` is creating a new copy of `*slog.Logger` and returns it, but the returned values is not used here, it just logs the error and then throw away the newly created logger, this doesn't make sense, just log it right away. Revert these changes everywhere please.
```suggestion
			log.Error("redis counter error", slog.Any("error", err))
```